x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -311,6 +331,48 @@ private void getVersionedDocument(String uid, ActionListener<VersionedDocument>
         });
     }
 
+    private void getVersionedDocuments(Collection<String> uids, ActionListener<Tuple<List<ProfileDocument>, List<String>>> listener) {
+        if (uids.isEmpty()) {
+            listener.onResponse(new Tuple<>(List.of(), List.of()));
+            return;
+        }
+        tryFreezeAndCheckIndex(listener).ifPresent(frozenProfileIndex -> {
+            frozenProfileIndex.checkIndexVersionThenExecute(
+                listener::onFailure,
+                () -> new OriginSettingClient(client, SECURITY_PROFILE_ORIGIN).prepareMultiGet()
```

Comment:
Since #86345 is now merged, we'd want 
```suggestion
                () -> new OriginSettingClient(client, getActionOrigin()).prepareMultiGet()
```